### PR TITLE
Release 4.3.0

### PR DESCRIPTION
**Pull request by joaander**
## Description

Changes needed to release 4.3.0.

## Motivation and context

Make the latest changes available to users.

## How has this been tested?

- [x] hoomd-bechmarks check for performance regressions
- [x] hoomd-validation check for correct results

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
hoomd-benchmarks results using hoomd-benchmarks@4213ab0869db211b867588a86b17c8da8facccd4

A100 GPU:
```
                                      4.1.0         4.2.0         4.3.0
HPMCSphere                     1.705589e+03  1.708129e+03  1.703098e+03
MDPairLJ                       4.494534e+03  4.463146e+03  4.498513e+03
MDPairOPP                      2.278491e+03  2.276466e+03  2.279397e+03
MDPairTable                    3.343676e+03  3.332324e+03  3.343470e+03
MDPairWCA                      9.431316e+03  9.318948e+03  9.402811e+03
MicrobenchmarkBoxResize        1.078090e+05  1.015943e+05  1.025325e+05
MicrobenchmarkEmptySimulation  2.016352e+07  2.003871e+07  2.001184e+07
MicrobenchmarkCustomTrigger    3.206513e+06  3.283930e+06  3.237147e+06
MicrobenchmarkCustomUpdater    6.667503e+06  6.850368e+06  7.160774e+06
MicrobenchmarkCustomForce      0.000000e+00  0.000000e+00  0.000000e+00
MicrobenchmarkGetSnapshot      6.319492e+02  6.549541e+02  6.040169e+02
MicrobenchmarkSetSnapshot      9.819957e+01  9.037300e+01  9.326960e+01
GSD                            5.062075e+02  5.157455e+02  5.112744e+02
GSDLog                         1.117512e+05  1.094011e+05  1.104155e+05
```
AMD EPYC 7763 (16 cores used)
```
                                      4.1.0         4.2.0         4.3.0
HPMCSphere                     1.842938e+02  1.811565e+02  1.843968e+02
MDPairLJ                       1.559762e+02  1.580454e+02  1.507605e+02
MDPairOPP                      4.518236e+01  4.501089e+01  4.246185e+01
MDPairTable                    1.333896e+02  1.338858e+02  1.295813e+02
MDPairWCA                      6.687741e+02  6.616365e+02  6.739758e+02
MicrobenchmarkBoxResize        1.210046e+04  1.206171e+04  1.224395e+04
MicrobenchmarkEmptySimulation  1.490755e+07  1.438382e+07  1.494697e+07
MicrobenchmarkCustomTrigger    3.251599e+06  3.158292e+06  3.028139e+06
MicrobenchmarkCustomUpdater    6.657760e+06  7.030701e+06  6.550868e+06
MicrobenchmarkCustomForce      2.747931e+03  3.243962e+03  3.181613e+03
MicrobenchmarkGetSnapshot      1.647334e+01  1.645758e+01  1.637234e+01
MicrobenchmarkSetSnapshot      1.299138e+01  1.302726e+01  1.312886e+01
GSD                            2.796517e+02  2.968959e+02  2.929203e+02
GSDLog                         2.663413e+04  2.615757e+04  2.562741e+04
```

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -24,8 +18,16 @@ Release checklist:
...
 - [ ] Run [hoomd-benchmarks](https://github.com/glotzerlab/hoomd-benchmarks), check for performance
   regressions with the previous release, and post the tables in the release pull request.
+- [ ] Run [hoomd-validation](https://github.com/glotzerlab/hoomd-validation).
+
+All releases:
+
+- [ ] Update change log.
+  - ``git log --format=oneline --first-parent `git log -n 1 --pretty=format:%H -- CHANGELOG.rst`...``
+  - [milestone](https://github.com/glotzerlab/hoomd-blue/milestones)
+- [ ] Check readthedocs build, especially change log formatting.
```
Comment (b-butler) [inline .github/ISSUE_TEMPLATE/release.md:28]: Did you want to remove build status and output?

**Review by b-butler — APPROVED** [review]
Left one comment to checkout otherwise looks good.

File: .github/ISSUE_TEMPLATE/release.md (hunk 1 above)
Comment (joaander) [reply .github/ISSUE_TEMPLATE/release.md:28]: Yes, this is intentional. This was replaced by the readthedocs integration in the PR Checks section long ago.

**Pull request merged**